### PR TITLE
[fix](schema change) fix delete predicate incorrect comparisons result with empty strings during schema change

### PR DESCRIPTION
**Pull request by luwei16**
### problem 

```
insert into t1 (k1,  k2,  v1) value(1, '', 2);
delete form t1 where k1 = 1 and k2 = '';
alter table modify column v1 string
select * from t1 //  expect 0 rows in return, but get 1 row.
```

### CODE REVIEW

**Comment by luwei16** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

File: regression-test/data/schema_change_p0/test_schema_change_delete.out
Comment (gavinchou) [inline regression-test/data/schema_change_p0/test_schema_change_delete.out:1]: no need  to add an empty file, just use `result = sql """` to check result.

File: regression-test/suites/schema_change_p0/test_schema_change_delete.groovy [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    sql """ delete from ${tableName} where col1 = -100 and col2 = '' and col3 = '2023-10-19' """
+
+    sql """ alter table ${tableName} MODIFY COLUMN `col5` string REPLACE NOT NULL """
+
+    waitForSchemaChangeDone {
+        sql """ SHOW ALTER TABLE COLUMN WHERE TableName='${tableName}' ORDER BY createtime DESC LIMIT 1 """
+        time 600
+    }
+
+    qt_select "select * from ${tableName}"
```
Comment (gavinchou) [inline regression-test/suites/schema_change_p0/test_schema_change_delete.groovy:64]: use `result = sql """` to get rid of empty .out file.

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by luwei16** [comment]
run p0

**Comment by luwei16** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**